src/content/docs/en/guides/astro-db.mdx

```diff
@@ -499,3 +499,21 @@ export default async function() {
 
 The value returned by `asDrizzleTable('Pets', Pets)` is equivalent to `import { Pets } from 'astro:db'`, but is available even when Astro’s type generation can’t run.
 You can use it in any integration code that needs to query or insert into the database.
+
+## Self-hosted production deployment
+
+:::caution
+This is an advanced feature that requires deploying without overriding an existing database. For a fully managed solution, try [connect to the Astro Studio platform](/en/recipes/studio/) instead.
+:::
+
+If you are deploying to a self-managed host such as a [virtual private server](https://en.wikipedia.org/wiki/Virtual_private_server) and do not want to connect to Astro Studio, you can connect to your database using the `ASTRO_DATABASE_FILE` environment variable. This pattern __does not__ work well with serverless hosts, as data is not retained in those environments.
+
+This path should be to your `.db` file on the host. It's important that you *do not* override this file on deployment, ever, or you will lose your production data. Additionally, any schema migrations must be managed manually using this method.
+
+If you're comfortable with the risks and are able to manage deployment yourself, you can set the `ASTRO_DATABASE_FILE` variable during your build and omit:
+
+```shell
+ASTRO_DATABASE_FILE=/srv/files/database.db astro build
+```
+
+The build will statically compile with this path as your production database. When you deploy and launch your server it will connect to the file at this path on the production host.
```

Comment:
Moved some things about and tried to clarify other things. I think it's probably OK to say explicitly that this method won't work with serverless environments.

> Additionally, any schema migrations must be managed manually using this method.

I wasn't 100% sure what this meant in the context, so I've left it as-is, along with the surrounding content. Does this essentially mean you need to run your migration in your project directory on the server, with the `ASTRO_DATABASE_FILE` configured in `.env`.

``````suggestion
## Self-hosted production deployment

If you deploy your site to a self-managed host such as a [virtual private server](https://en.wikipedia.org/wiki/Virtual_private_server), you can choose to use a database file instead of connecting to Astro Studio.

:::caution
Using a database file is an advanced feature, and care should be taken when deploying to prevent overriding your database. This method will also not work in serverless deployments, as the file system is not persisted in those environments. For a fully managed solution, try [connect to the Astro Studio platform](/en/recipes/studio/) instead.
:::

To use a database file, set the `ASTRO_DATABASE_FILE` environment variable. This path should point to your `.db` file within the host environment. It's important that you *do not* override this file on deployment, ever, or you will lose your production data. Additionally, any schema migrations must be managed manually using this method.

If you're comfortable with the risks and are able to manage deployment yourself, you can set the `ASTRO_DATABASE_FILE` variable during your build and omit:

```shell
ASTRO_DATABASE_FILE=/srv/files/database.db astro build
```

The build will statically compile with this path as your production database. When you deploy and launch your server it will connect to the file at this path on the production host.
``````

Reply:
I like it!